### PR TITLE
Transfer must be inserted into the db at all times

### PR DESCRIPTION
**Pull request by matszczygiel**

### CODE REVIEW

File: drop-transfer/src/ws/client/mod.rs [hunk 1]
```diff
@@ -223,8 +223,8 @@ impl RunContext<'_> {
         let (tx, rx) = mpsc::unbounded_channel();
 
         let mut lock = self.state.transfer_manager.lock().await;
-        lock.insert_transfer(self.xfer.clone(), TransferConnection::Client(tx))
-            .await?;
+        lock.insert_transfer(self.xfer.clone(), TransferConnection::Client(tx))?;
+        drop(lock);
```
Comment (Lipt0nas) [inline drop-transfer/src/ws/client/mod.rs:227]: Is there a reason this can't be a one-liner?

**Review by LukasPukenis — COMMENTED** [review]
It looks like now the TransferManager and Storage receive the transfer at different times. TransferManager inserting to storage is less error prone, can it be tweaked this way?

**Comment by matszczygiel** [comment]
@LukasPukenis the close coupling was exactly the reason the issue was there in the first place

File: drop-transfer/src/ws/client/mod.rs (hunk 1 above)
Comment (matszczygiel) [reply drop-transfer/src/ws/client/mod.rs:227]: Probaably

Comment (matszczygiel) [reply drop-transfer/src/ws/client/mod.rs:227]: done

**Comment by LukasPukenis** [comment]
@matszczygiel what I mean is that manager knows about the transfer earlier than the storage or vice-versa so now it might manifest in another race condition down the line as they're tightly related

File: drop-transfer/src/ws/client/mod.rs (hunk 1 above)
Comment (Lipt0nas) [reply drop-transfer/src/ws/client/mod.rs:227]: I think you forgot to add the changes before pushing :sweat_smile:

**Review by LukasPukenis — APPROVED** [review]
Looks solid

**Pull request merged**